React/CoreModules/RCTDeviceInfo.mm

```diff
@@ -174,21 +175,31 @@ - (void)interfaceOrientationDidChange
 - (void)_interfaceOrientationDidChange
 {
   UIInterfaceOrientation nextOrientation = [RCTSharedApplication() statusBarOrientation];
+  UIApplicationState appState = [RCTSharedApplication() applicationState];
+  BOOL isRunningInFullScreen = CGRectEqualToRect([UIApplication sharedApplication].delegate.window.frame, [UIApplication sharedApplication].delegate.window.screen.bounds);
+    
+  
+  BOOL isActive = appState == UIApplicationStateActive;
+  BOOL isChangingFullscreen = (isRunningInFullScreen != _isFullscreen || !isRunningInFullScreen);
```

Comment:
this is a weird condition, and the name does not seem to match the condition
If we are matining the condition, we need a comment to explainf why this is not simply 
```BOOL isChangingFullscreen = isRunningInFullScreen != _isFullscreen;```
The condition could be simplified to
```BOOL isChangingFullscreen = !isRunningInFullScreen || !_isFullscreen;```
And again I'm not entirely sure that condition makes sense

Reply:
This condition indeed can be shortened to what you suggested, good catch. Thanks.

The condition is needed and called this way because it catches two scenarios:

a) Simple resize when the app isn't in the fullscreen mode right now - the !isRunningInFullScreen
b) Resize when the app gets from slide over mode to fullscreen - isRunningInFullScreen != _isFullscreen

Anyway, I commented this line with the above.